tailscale/rootfs/etc/s6-overlay/s6-rc.d/tailscaled/run

```diff
@@ -30,15 +33,49 @@ then
   options+=(--tun=userspace-networking)
 fi
 
+# Prepare DSCP setting
+if ! bashio::config.has_value "dscp" || \
+  bashio::config.equals "dscp" 0;
+then
+  tailscaled_gid=0
+else
+  tailscaled_gid=${TAILSCALED_GID}
+
+  # It is not strictly necessary to create the group for the gid, s6-setuidgid can set any arbitrary value, but at least we can identify it with a name
+  # Until processes running with root user, they won't have restrictions
+  addgroup -g ${TAILSCALED_GID} -S tailscaled || true
+  adduser root tailscaled || true
+
+  bashio::log.info "Setting DSCP for tailscaled (IPv4)"
+  if (( 0 < $( \
+    iptables -t mangle -S OUTPUT \
+    | { grep -Ec "^-A OUTPUT -m owner --gid-owner ${TAILSCALED_GID} -j DSCP --set-dscp" || true ;}) ))
+  then
+    bashio::log.notice "  DSCP is already set for tailscaled"
+  elif ! iptables -t mangle -A OUTPUT -m owner --gid-owner ${TAILSCALED_GID} -j DSCP --set-dscp $(bashio::config "dscp"); then
+    bashio::log.warning "  Setting DSCP for tailscaled is unsuccessful"
+  fi
+
+  bashio::log.info "Setting DSCP for tailscaled (IPv6)"
+  if (( 0 < $( \
+    ip6tables -t mangle -S OUTPUT \
+    | { grep -Ec "^-A OUTPUT -m owner --gid-owner ${TAILSCALED_GID} -j DSCP --set-dscp" || true ;}) ))
+  then
+    bashio::log.notice "  DSCP is already set for tailscaled"
+  elif ! ip6tables -t mangle -A OUTPUT -m owner --gid-owner ${TAILSCALED_GID} -j DSCP --set-dscp $(bashio::config "dscp"); then
+    bashio::log.warning "  Setting DSCP for tailscaled is unsuccessful"
+  fi
+fi
+
 # Run Tailscale
 if bashio::debug ; then
-  exec /opt/tailscaled "${options[@]}"
+  exec s6-setuidgid 0:${tailscaled_gid} /opt/tailscaled "${options[@]}"
```

Comment:
This goes beyond what is recommend in HA. We shouldn't be running this stuff in random GIDs or UIDs